extensions/amp-intersection-observer-polyfill/0.1/amp-intersection-observer-polyfill.js

```diff
@@ -0,0 +1,28 @@
+
+import {getService, registerServiceBuilder} from '../../../src/service';
+import {upgradePolyfill} from '../../../src/polyfills/intersection-observer';
+
+const TAG = 'amp-intersection-observer-polyfill';
+
+
+/**
+ * @param {!Window} win
+ */
+function install(win) {
+  upgradePolyfill(win, () => {
+    // QQQQQQ: install polyfill.
+    win.IntersectionObserver = win.QqqqInOb;
+    win.IntersectionObserverEntry = win.QqqqInObEntry;
```

Comment:
I don't think we need to touch `Entry` constructor. The primary API is just the `IntersectionObserver` constructor, then the entry instances passed to the callback.

Reply:
Agree.